### DISCUSSION TITLE
Assigning an interrupt to a timer

### DISCUSSION THREAD

**Discussion by PDE-Paul** [espressif/arduino-esp32#9982] [Q&A]
Hi,

Can anyone please provide me with some guidance regarding the V3.0.X changes to the Timer API.

Previous versions of the API, I setup 4 separate hardware timers each running a separate critical routines when triggered, now I have no idea how to "force" assign what timer does what routine.

Previous code was...
timer = timerBegin(0, 80, true); //(Timer 0)
.....

BLEtimer = timerBegin(1, 80, true); //(Timer 1)
.....

FBPtimer = timerBegin(2, 80, true); //(Timer 2)
........

AZPtimer = timerBegin(3, 80, true); //(Timer 3)

Now if I try...
timer = timerBegin(1000000);  // how do I force this to be Timer 0?
.....

BLEtimer = timerBegin(1000000);  // how do I force this to be Timer 1?
.....

FBPtimer = timerBegin(10000);  // how do I force this to be Timer 2?
....

AZPtimer = timerBegin(1000000);  // how do I force this to be Timer 3?
.....

Is there an undocumented trick to this? Or are Timers automatically assigned as they are setup and initialized? 

Any assistance would be greatly appreciated.

Kind Regards
Paul

**Comment by me-no-dev** [comment]
Why do you need to force particular timer number? Does it matter which number it is? You will not get more than the available timers anyway.

**Reply by PDE-Paul** [reply]
Hi, 

I use all 4 Hardware timers and only 4 Timers. 

I have worked out the API auto assigns the Timers, so far I have working code, just working out all the other migration changes. 

Thank you for the comment. 
Regards Paul
